src/shared/components/query/QueryStore.ts

```diff
@@ -563,6 +504,194 @@ export class QueryStore
+
+			const _vs = {...this.userVirtualStudiesSet.result, ...this.sharedVirtualStudiesSet.result};
+
+			const physicalStudyIds = _.filter(this._allSelectedStudyIds.keys(),studyId=> _vs[studyId] === undefined);
+
+			if(this._allSelectedStudyIds.keys().length !== physicalStudyIds.length){
+				await Promise.all(_.map(physicalStudyIds, studyId => {
+					return client.getAllSamplesInStudyUsingGET({
+						studyId: studyId
+					}).then(samples => {
+						studyToSampleSet[studyId] = stringListToSet(samples.map(sample => sample.sampleId));
+					})
+				}));
+	
+				this._allSelectedStudyIds.keys().forEach(id=>{
+					if(_vs[id]){
+						let virtualStudy = _vs[id];
+						virtualStudy.data.studies.forEach(study => {
+							if(studyToSampleSet[study.id] === undefined){
+								studyToSampleSet[study.id] = stringListToSet(study.samples)
+							}
+						});
+					}
+				});
+			} else {
+				physicalStudyIds.forEach(studyId=>{
+					studyToSampleSet[studyId] = {}
+				})
+			}
+			return studyToSampleSet;
+		},
+		default: {}
+	})
+
+	private readonly selectableStudiesSet = remoteData<{[studyId:string]:string[]}>({
+		await: ()=>[this.physicalStudiesSet, this.selectedStudyToSampleSet],
+		invoke: async ()=>{
+			let result:{[studyId:string]:string[]} = {}
+		result = _.reduce(this.physicalStudiesSet.result,(acc, next, key)=>{
+			acc[key] = [key]
+			return acc;
+		},result)
+
+		result = _.reduce(this.userVirtualStudiesSet.result,(acc, next, key)=>{
+			acc[key] = [key]
+			return acc;
+		},result)
+
+		result = _.reduce(this.sharedVirtualStudiesSet.result,(acc, next, key)=>{
+			acc[key] = next.data.studies.map(study=> study.id)
+			return acc;
+		},result)
+		return  result;
+		},
+		default: {},
+		onResult:(selectableStudiesSet)=>{
+			if(this.selectedSampleListId !== CUSTOM_CASE_LIST_ID) {
+				let _userSelectableIds = [..._.keys(this.physicalStudiesSet.result), ..._.keys(this.userVirtualStudiesSet.result)];
+				let sharedIds:string[] = [];
+				let unknownIds:string[] = [];
+
+				this._allSelectedStudyIds.keys().forEach(id=>{
+					if(selectableStudiesSet[id]){
+						if(!_.includes(_userSelectableIds, id)){
+							sharedIds.push(id)
+						}
+					}else{
+						unknownIds.push(id);
+					}
+				});
+				//this block is executed when the query is a saved virtual study query is shared to other user
+				//in this scenario we override some parameters to correctly show selected cases to user
+				if(!_.isEmpty(sharedIds) && _.isEmpty(unknownIds)){
+					let studySampleMap = this._defaultStudySampleMap
+					if(studySampleMap === undefined || _.isEmpty(studySampleMap)){
+						studySampleMap = _.reduce(this.selectedStudyToSampleSet.result,(acc, next, studyId)=>{
+							acc[studyId] = _.keys(next)
+							return acc;
+						},{} as {[id: string]: string[]})
+					}
+
+					let isStudySampleMapEmpty = true;
+					_.each(studySampleMap,(samples) => {
+						if(samples.length > 0 ){
+							isStudySampleMapEmpty = false;
+							return false;

```

Comment:
returning false from _.each won't break looping.    this looks to me like a case for using _.some